### PR TITLE
add sso tests

### PR DESCRIPTION
**Pull request by orthagh**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

WIP, can't test locally

### CODE REVIEW

File: tests/LDAP/AuthLdap.php [hunk 1]
```diff
@@ -765,4 +765,38 @@ public function testRemovedUser() {
...
+      return $sso_vars;
+   }
+
+   /**
+    * @dataProvider ssoVariablesProvider
+    */
+   public function testOtherAuth($sso_field_id, $sso_field_name) {
+      //$this->dump($sso_field_id, $sso_field_name);
+      $config = new \Config;
+      $config->update([
```
Comment (trasher) [inline tests/LDAP/AuthLdap.php]: Check upate result ;)

File: tests/LDAP/AuthLdap.php [hunk 2]
```diff
@@ -765,4 +765,38 @@ public function testRemovedUser() {
...
+         'id' => 1,
+         'ssovariables_id' => $sso_field_id,
+      ]);
+
+      $_SERVER[$sso_field_name] = 'ecuador0';
+
+      $this->dump($sso_field_name, $_SERVER[$sso_field_name]);
+
+      unset($_SESSION['glpiname']);
```
Comment (trasher) [inline tests/LDAP/AuthLdap.php]: `$_SERVER[$sso_field_name]` no?

Comment (orthagh) [reply tests/LDAP/AuthLdap.php]: nop, just removing a little var in session to check if correctly set after login method

Comment (trasher) [reply tests/LDAP/AuthLdap.php]: ok :)

**Comment by trasher** [comment]
Could probably be cherry-picked on 9.2/bf branch

File: tests/LDAP/AuthLdap.php [hunk 3]
```diff
@@ -765,4 +765,35 @@ public function testRemovedUser() {
...
+      return $sso_vars;
+   }
+
+   /**
+    * @dataProvider ssoVariablesProvider
+    */
+   public function testOtherAuth($sso_field_id, $sso_field_name) {
+      global $CFG_GLPI;
+
+      $config_values = \Config::setConfigurationValues('core', [
```
Comment (trasher) [inline tests/LDAP/AuthLdap.php]: Maybe conf should be reset after tests? Since value in database is changed; this may cause issues in the future.

Comment (orthagh) [reply tests/LDAP/AuthLdap.php]: done

**Pull request merged**